tests/test_views_list_post.py

```diff
@@ -140,3 +140,29 @@ def test_validation_error(app, db, test_data, search_url, content_type):
         res = client.post(
             search_url, data=json.dumps(test_data[0]), headers=HEADERS)
         assert res.status_code == 400
+
+
+@mock.patch('invenio_records.api.Record.validate', _mock_validate_fail)
```

Comment:
If this is here, doesn't it mean that the JSON schema is actually not being validated? I mean if you remove, I think it should still fail, and it would test better what you need.